Packs/PAN-OS/Integrations/Panorama/Panorama.py

```diff
@@ -881,29 +883,81 @@ def template_test(template: str):
                         f' The available Templates for this instance: {", ".join(template_names)}.')
 
 
-@logger
+@polling_function(
+    name=demisto.command(),
+    interval=60,
+    timeout=1200
+)
+def pan_os_command_polling(args: dict, result):
+    """
+    Polls job status until it is done
+    """
+    job_id = args.get("job-id")
+    status = result.get('response', {}).get('result', {}).get('job', {}).get('status', '')
+    if not job_id:
+        raise DemistoException("job-id is required when polling=true.")
+
+    context_output = {
+        'JobID': job_id,
+        'Status': status
+    }
+    continue_to_poll = True
+    res = {
+        'Type': entryTypes['note'],
+        'ContentsFormat': formats['json'],
+        'Contents': result,
+        'ReadableContentsFormat': formats['text'],
+        'HumanReadable': f'Job status: {status}',
+    }
+    return PollResult(
+        response=res,
+        continue_to_poll=status != 'FIN',  # continue polling if job isn't done
+        partial_result=CommandResults(
+            readable_output=f'Waiting for Job-ID {job_id} to finish...'
+        )
+    )
+
+
 def panorama_command(args: dict):
     """
     Executes a command
     """
     params = {}
+    is_xml = False
+    is_file = False
+    polling = False
     for arg in args.keys():
-        params[arg] = args[arg]
+        if arg == "is_xml":
+            is_xml = argToBoolean(args[arg])
+        elif arg == "is_file":
+            is_file = argToBoolean(args[arg])
+        elif arg == "polling":
+            polling = argToBoolean(args[arg])
+        elif arg == "file_name":
+            pass
```

Comment:
```suggestion
```

redundant

Reply:
Not sure if I'm tracking but I tried to implement this change, lmk if it is what you had in mind.